### PR TITLE
[13.0] [MIG] sale_order_product_assortment: Migration to v13.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT27472

Please @joao-p-marques  @chienandalu  review this.

### CODE REVIEW

File: sale_order_product_assortment/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        string="Blacklisted Products",
+        compute="_compute_product_assortment_ids",
+    )
+    has_whitelist = fields.Boolean(compute="_compute_product_assortment_ids")
+    has_blacklist = fields.Boolean(compute="_compute_product_assortment_ids")
+
+    @api.depends("partner_id")
+    def _compute_product_assortment_ids(self):
+        # If we don't initialize the fields we get an error with NewId
+        self.whitelist_product_ids = self.env["product.product"]
```
Comment (chienandalu) [inline sale_order_product_assortment/models/sale_order.py:25]: Don't you loop over the recorset?

Comment (CarlosRoca13) [reply sale_order_product_assortment/models/sale_order.py:25]: In this case I want to initialize the field as a empty recordset. I don't know if this is the correct way

Comment (joao-p-marques) [reply sale_order_product_assortment/models/sale_order.py:25]: If that's what you want, that seems like the correct way, according to the [documentation](https://www.odoo.com/documentation/13.0/reference/orm.html#environment).
However, I don't really understand the logic... AFAICS, you are reassigning it [bellow](https://www.odoo.com/documentation/13.0/reference/orm.html#environment), right? What was the error with `NewId` you refer to in the comment?

Comment (chienandalu) [reply sale_order_product_assortment/models/sale_order.py:25]: But what about later?

Comment (CarlosRoca13) [reply sale_order_product_assortment/models/sale_order.py:25]: I just want to have records if a partner is defined... so here you have were we add the records https://github.com/OCA/sale-workflow/pull/1407/files#diff-d857107550de88517b0f4cb32ae7770a25a630dc254b97191e7f71f262183089R52-R57

Comment (chienandalu) [reply sale_order_product_assortment/models/sale_order.py:25]: Can you write over multiple records in v13 with a direct assignment?

Comment (CarlosRoca13) [reply sale_order_product_assortment/models/sale_order.py:25]: @joao-p-marques that's the error that i catch if I don't use this...
image

Comment (joao-p-marques) [reply sale_order_product_assortment/models/sale_order.py:25]: Ok, does it give any more information in the logs?

**Review by chienandalu — APPROVED** [review]

**Comment by chienandalu** [comment]
I resolved my doubt :)

File: sale_order_product_assortment/models/sale_order.py (hunk 1 above)
Comment (CarlosRoca13) [reply sale_order_product_assortment/models/sale_order.py:25]: @joao-p-marques the logs just give me this info `2021-01-04 09:43:48,203 1 WARNING devel odoo.http: ('sale.order(<NewId 0x7f7c50380ef0>,).whitelist_product_ids', None)`

Comment (joao-p-marques) [reply sale_order_product_assortment/models/sale_order.py:25]: Ok. Your latest push also cleared it for me :smile: Good job!

**Review by joao-p-marques — APPROVED** [review]

**Review by victoralmau — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]

**Pull request merged**